### PR TITLE
feat(ToggleGroup): ✨ New `ToggleGroup` component

### PR DESCRIPTION
**Pull request by Magnusrm**
### ToggleGroup

- Adds a new `ToggleGroup` component which can be populated by `ToggleGroup.Item`s and allow for one of the items to be active by handling the value of the `ToggleGroup`. 
- The `ToggleGroup`s value can be both controlled using state and uncontrolled. 
- When using the ToggleGroup uncontrolled, it is recommended to set a default value using the property `defaultValue`.
- The `ToggleGroup` is using `RovingTabindexRoot` as a `div`, and the `ToggleGroup.Item` is using `RovingTabindexItem` as a designsysstem `Button`.

Roving Tabindex:
- Tab navigation in the `ToggleGroup` is handled by the new `RovingTabindex` utility component. 
- The RovingTabindex feature allows for wrapping a bunch of the extensible component `RovingTabindexItem` in the extensible component `RovingTabindexRoot`. You can then tab in and out of the `RovingTabindexRoot` and only one of the items will be focused.
- You can also move focus left and right using the arrow keys.

Example usage:
```
import { ToggleGroup } from '@digdir/design-system-react';

<ToggleGroup defaultValue="value1">
  <ToggleGroup.Item value="value1">Option 1</ToggleGroup.Item>
  <ToggleGroup.Item value="value2">Option 2</ToggleGroup.Item>
  <ToggleGroup.Item value="value3">Option 3</ToggleGroup.Item>
</ToggleGroup>;

```

Resolves: #805

### CODE REVIEW

**Comment by mrosvik** [comment]
Great work! 👏🏻 🤩
 
 I think..
- `Focus`-state should behave like the `radio` component (only show with tab-focus).
- We should add an example with only icons and make sure the buttons are square.

image

**Comment by Magnusrm** [comment]
> * `Focus`-state should behave like the `radio` component (only show with tab-focus).
> * We should add an example with only icons and make sure the buttons are square.

@mrosvik 
I agree on adding example with only icons, nice catch! 😊 I'll also update the focus behaviour 👍

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  defaultValue?: string;
+  onChange?: (value: string) => void;
+  name?: string;
+  size?: 'small' | 'medium' | 'large';
+};
+
+export const ToggleGroupContext = createContext<ToggleGroupContextProps>({});
+
+export type ToggleGroupProps = {
+  /** Value of the ToggleGroup component. Set when you want to controll the value of the component. */
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:20]:
```suggestion
  /** Controlled state for `ToggleGroup` component. */
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  value?: string;
+  /** Default value of the ToggleGroup component. Set when you want to use the component as an uncontrolled component. */
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:22]:
```suggestion
  /** Default value. */
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  defaultValue?: string;
+  /** Function that is called when the value of the ToggleGroup component changes. The current value is available in the callback. */
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:24]:
```suggestion
  /** Callback with selected `ToggleGroupItem` `value` */
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  onChange?: (value: string) => void;
+  /** Name of the ToggleGroup component. */
+  name?: string;
+  /** Size of the ToggleGroup component. */
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:28]:
```suggestion
  /** Changes items size and paddings */
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+export const ToggleGroupContext = createContext<ToggleGroupContextProps>({});
+
+export type ToggleGroupProps = {
+  /** Value of the ToggleGroup component. Set when you want to controll the value of the component. */
+  value?: string;
+  /** Default value of the ToggleGroup component. Set when you want to use the component as an uncontrolled component. */
+  defaultValue?: string;
+  /** Function that is called when the value of the ToggleGroup component changes. The current value is available in the callback. */
+  onChange?: (value: string) => void;
+  /** Name of the ToggleGroup component. */
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:26]:
```suggestion
  /** Form element name */
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    let onValueChange = onChange;
+    if (!isControlled) {
+      onValueChange = (newValue: string) => {
+        setUncontrolledValue(newValue);
+        onChange?.(newValue);
+      };
+      value = uncontrolledValue;
+    }
+    const nameId = useId();
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:50]: nitpicking, but suggest moving this to line 37 for better readability of code

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      <div
+        {...rest}
+        className={cn(classes.toggleGroupContainer, rest.className)}
+        ref={ref}
+      >
+        <ToggleGroupContext.Provider
+          value={{
+            value,
+            defaultValue,
+            name: name ?? `radiogroup-name-${nameId}`,
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:62]: We should use the name of the component as thats where to data comes from when form is submitted.
```suggestion
            name: name ?? `togglegroup-name-${nameId}`,
```

File: packages/react/src/components/ToggleGroup/ToggleGroupItem/useToggleGroupitem.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+/** Handles props for `ToggleGroup.Item` in context with `ToggleGroup` and `RovingTabIndex` */
+export const useToggleGroupItem: UseToggleGroupItem = (
+  props: ToggleGroupItemProps,
+) => {
+  const { ...rest } = props;
+  const toggleGroup = useContext(ToggleGroupContext);
+  const itemValue =
+    props.value ?? (typeof props.children === 'string' ? props.children : '');
+  const active = toggleGroup.value == itemValue;
+  const buttonId = `toggleButton-${useId()}`;
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroupItem/useToggleGroupitem.ts:26]:
```suggestion
  const buttonId = `togglegroup-item-${useId()}`;
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+      <ToggleGroup.Item>Peanut</ToggleGroup.Item>
+      <ToggleGroup.Item>Walnut</ToggleGroup.Item>
+      <ToggleGroup.Item>Pistachio 🤤</ToggleGroup.Item>
+    </ToggleGroup>
+  );
+};
+
+Preview.args = {
+  size: 'medium',
+  defaultValue: 'test',
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx:49]: Suggest removing this default value as it does not match the example used in preview.
```suggestion
  defaultValue: 'test',
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+  const myGroupRef = React.createRef<HTMLDivElement>();
+  const myItemRef = React.createRef<HTMLButtonElement>();
+  return (
+    <>
+      <div style={{ display: 'flex', gap: '4px' }}>
+        <Button
+          size='small'
+          onClick={() => myItemRef.current?.focus()}
+        >
+          Focus last item!
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx]: Suggest using the actual name/value for item for better readability.

```suggestion
          Focus Walnut
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        </Button>
+        <Button
+          size='small'
+          onClick={() => setValue('peanut')}
+        >
+          Activate middle button!
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx:104]:
```suggestion
          Select Peanut
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.tsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+  /** Default value of the ToggleGroup component. Set when you want to use the component as an uncontrolled component. */
+  defaultValue?: string;
+  /** Function that is called when the value of the ToggleGroup component changes. The current value is available in the callback. */
+  onChange?: (value: string) => void;
+  /** Name of the ToggleGroup component. */
+  name?: string;
+  /** Size of the ToggleGroup component. */
+  size?: 'small' | 'medium' | 'large';
+} & Omit<HTMLAttributes<HTMLDivElement>, 'value' | 'onChange'>;
+
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.tsx:31]: Suggest adding a small jsdoc for the actual component. Look at `Textfield` for example

**Review by mimarz — CHANGES_REQUESTED** [review]
Oh boy! Nice work on the Roving utility component 👌 

Looks very good, mostly just feedback on some docs and clean code. 

ohboy-golden-girls

**Comment by mimarz** [comment]
Suggest also updating the [Overview.mdx](https://github.com/digdir/designsystem/blob/cf8343a872d4f19ca136f868a7e1bbebf01bee18/packages/Overview.mdx#L45C6-L45C6) to point to this new "felles" component :)

File: packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+export default {
+  title: 'Felles/ToggleGroup',
+  component: ToggleGroup,
+} as Meta;
+
+export const Preview: StoryFn<typeof ToggleGroup> = (args) => {
+  return (
+    <ToggleGroup
+      {...args}
+      defaultValue='Peanut'
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx:38]: For our Preview's always only use `args` for defining prop values so its clear to our users in the props table that this is defined without needing to inspect code.

File: packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+    <ToggleGroup
+      defaultValue={'Pistachio'}
+      onChange={handleChange}
+    >
+      <ToggleGroup.Item icon={<AkselIcon />}>Pistachio</ToggleGroup.Item>
+      <ToggleGroup.Item icon={icon}>Peanut</ToggleGroup.Item>
+      <ToggleGroup.Item icon={<AkselIcon2 />}>Walnut</ToggleGroup.Item>
+    </ToggleGroup>
+  );
+};
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx:92]: No need for a separate story for uncontrolled as `Preview` already shows this. 
```suggestion
```

File: packages/react/src/components/ToggleGroup/ToggleGroup.mdx [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+</ToggleGroup>;
+```
+
+## Only Icons
+
+<Canvas of={ToggleGroupStories.OnlyIcons} />
+
+## Uncontrolled
+
+<Canvas of={ToggleGroupStories.Uncontrolled} />
```
Comment (mimarz) [inline packages/react/src/components/ToggleGroup/ToggleGroup.mdx:35]: No need for a separate story for uncontrolled as Preview already shows this.
```suggestion
```

**Review by mimarz — APPROVED** [review]
Some more suggestions on docs else looking good! 👍

**Pull request merged**